### DISCUSSION TITLE
Only root-level navigation destinations are effective for a navigation stack with a homogeneous path.SwiftUI.AnyNavigationPath.Error.comparisonTypeMismatch

### DISCUSSION THREAD

**Discussion by Cu-Toof** [pointfreeco/swift-composable-architecture#2349] [Beta]
I create a flow UI :

```
- NavigationStackStore
   - Splash
   - TabView
      - Home: NavigationStackStore
      - Settings: NavigationStackStore
```

When launching the app, I receive an error: 
```
Only root-level navigation destinations are effective for a navigation stack with a homogeneous path.
SwiftUI/NavigationColumnState.swift:508: Fatal error: 'try!' expression unexpectedly raised an error: SwiftUI.AnyNavigationPath.Error.comparisonTypeMismatch
```

**Comment by mbrandonw** [comment]
Hi @Cu-Toof, it is not possible to diagnose this without looking at some actual code. Ideally you could share a very small project that reproduces the problem so that we can run it ourselves. 

Most likely something in your code is misconfigured, or you have run into one of the many bugs that plagues vanilla SwiftUI.

**Reply by mbrandonw** [reply]
Although, it does seem like you are trying to push a tab view onto a navigation stack. Is that true? I don’t believe that is supposed in SwiftUI.

**Reply by Cu-Toof** [reply]
I did create a small project: https://github.com/Cu-Toof/DemoNavigationStackStore
@mbrandonw

**Reply by Cu-Toof** [reply]
@mbrandonw I'm trying to split the flow UI so that will be easy to manage and reduce effort to Root NavigationStack.

**Reply by mbrandonw** [reply]
Hi @Cu-Toof, the project you created does not compile. I believe it needs this change to be applied:

```diff
 CaseLet(
   /Home.Path.State.detail,
   action: Home.Path.Action.detail,
-  then: HomeDetail.init(store:)
+  then: HomeDetailView.init(store:)
 )
```

Further, the way you have set up your views does not work even in vanilla SwiftUI. Here is a vanilla reproduction of your demo:

<details>

<summary>
Click to expand
</summary>

```swift
import SwiftUI

class SplashModel: ObservableObject {
  @Published var path: [Path] = []
  enum Path: Hashable {
    case tabBar(TabBarModel)
  }
  func onAppear() {
    self.path.append(.tabBar(TabBarModel()))
  }
}

struct SplashView: View {
  @ObservedObject var model = SplashModel()

  var body: some View {
    NavigationStack(path: self.$model.path) {
      Form {
        NavigationLink(value: SplashModel.Path.tabBar(TabBarModel())) {
          Text("Go")
        }
      }
      .navigationDestination(for: SplashModel.Path.self) {
        switch $0 {
        case let .tabBar(model):
          TabBarView(model: model)
        }
      }
    }
    .onAppear {
      self.model.onAppear()
    }
  }
}

class TabBarModel: ObservableObject, Hashable {
  @Published var selectedTab = Tab.home
  @Published var home = HomeModel()
  @Published var settings = SettingsModel()
  enum Tab { case home, settings }

  static func == (lhs: TabBarModel, rhs: TabBarModel) -> Bool {
    lhs === rhs
  }
  func hash(into hasher: inout Hasher) {
    hasher.combine(ObjectIdentifier(self))
  }
}
struct TabBarView: View {
  @ObservedObject var model: TabBarModel

  var body: some View {
    TabView(selection: self.$model.selectedTab) {
      HomeView(model: self.model.home)
        .tabItem { Text("Home") }
      SettingsView(model: self.model.settings)
        .tabItem { Text("Settings") }
    }
  }
}

class HomeModel: ObservableObject {
  @Published var path: [Int] = []
}
struct HomeView: View {
  @ObservedObject var model: HomeModel
  var body: some View {
    NavigationStack(path: self.$model.path) {
      Button("Go") {
        self.model.path.append(.random(in: 1...1_000))
      }
      .navigationDestination(for: Int.self) {
        Text("\($0)")
      }
    }
  }
}
class SettingsModel: ObservableObject {
}
struct SettingsView: View {
  @ObservedObject var model: SettingsModel
  var body: some View {
    NavigationStack {
      Text("Settings")
    }
  }
}

#Preview {
  SplashView()
}
```
</details>

If you run that in a simulator you will see really strange behavior. It's not exactly the behavior you are seeing in your TCA demo, but it's enough to show that SwiftUI simply does not support pushing a tab view onto a navigation stack that itself has a nav stack embedded inside.

So I believe you just need to re-think the navigation structure of your app.

**Comment by g0dzillaa** [comment]
Hey @mbrandonw,

In this case, what would your recommendation be?

I am trying TCA for the first time and have encountered a similar situation. What I am trying to do is to break up the app into different parts (with different `NavigationStackStore` for each part) and then have a root `NavigationStackStore` to connect all the parts. Ideally, I got this thought so that I can utilise Tree Based and Stack Based in different parts of the app. 

However, as far as I understand in the vanilla Swift world, you should have only one NavigationStack and multiple`.navigationDestination`s.

**Reply by Cu-Toof** [reply]
Currently, I'm using SwitchStore to display Splash or TabView, but transition is not good (no animation). If you have any solutions, please tell me
